### PR TITLE
bugfixes for global state

### PR DESCRIPTION
**Pull request by bqpd**
And small fixes from Hoburg's review!

@whoburg, ready for review

@galbramc, ready for MOSEK 8

### CODE REVIEW

**Comment by galbramc** [comment]
test this please

**Comment by galbramc** [comment]
test this please

**Comment by galbramc** [comment]
test this please

**Comment by galbramc** [comment]
Ok you are all set with a failing Mosek 8 on just reynolds-ubuntu16.

**Comment by bqpd** [comment]
@galbramc, I can't reproduce the ubuntu bug on my ubuntu 16.06 machine, so could we revert to MOSEK 7 on that one for a bit and I'll make an issue?

**Comment by whoburg** [comment]
@galbramc, in addition to the MOSEK 7 revert request, we have an additional request.

In this PR there is a file `gpkit/tests/diff_output.py`. It can be run at the command line via the following horrible quote-filled bash syntax:
`python diff_output.py "`cat output1.txt`" "`output2.txt`"`

If the two files are the same or have only small differences in numerical values, it will return printing nothing. But if they differ, it will raise a RuntimeWarning.

We'd like you to modify the diff'ing you're currently doing to run diff_output.py instead. If it prints nothing, consider that a success. If it raises the runtime warning, then please run diff for us (so we can see the difference) and mark the failure.

This helps us with the problem that numerical outputs are sometimes slightly different several decimal places in (e.g. from MOSEK 7 to 8)

File: gpkit/tests/t_constraints.py [hunk 1]
```diff
@@ -199,12 +199,12 @@ def test_posyconstr_in_sp(self):
         y = Variable('y')
         with SignomialsEnabled():
             sig_constraint = (x + y >= 0.1)
-        m = Model(x, [Tight([x >= y], raiseerror=True),
-                      x >= 2, y >= 1, sig_constraint])
+        m = Model(x*y, [Tight([x >= y], raiseerror=True),
+                        x >= 2, y >= 1, sig_constraint])
```
Comment (whoburg) [inline gpkit/tests/t_constraints.py]: why the change to the cost?

**Comment by whoburg** [comment]
@bqpd, do we know why this is failing on reynolds-ubuntu16?

**Comment by galbramc** [comment]
Ok I'm a bit confused. I don't actually look for any differences right now when you run mosek. We only check for differences with cvxopt. Am I supposed to only run this diff when a file is modified? If the file was modified, then it was overwritten, and I don't have the original to compare with. I think I need a bit more info to add this diff script...

**Comment by galbramc** [comment]
These are the errors with Mosek 8:

```
ERROR [0.023s]: test_unbounded_debugging (gpkit.tests.t_model.TestSP_mosek)
Test nearly-dual-feasible problems
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/jenkins/workspace/gpkit_PullRequest_Unit/buildnode/reynolds-ubuntu16/optimizer/mosek/gpkit/tests/t_model.py", line 409, in test_unbounded_debugging
    m.solve(self.solver, verbosity=0)
AssertionError: (<type 'exceptions.RuntimeWarning'>, <type 'exceptions.ValueError'>) not raised
```

**Comment by whoburg** [comment]
@galbramc understood, we'll have to think carefully about how we want those tests to be run. Will get back to you.

For now, can we roll back the mosek 8 testing to mosek 7? I think this should then have passing tests. We're eager to merge it for our version 0.5 release.

**Comment by whoburg** [comment]
retest this please

**Comment by whoburg** [comment]
I just attempted to revert gpkit_PullRequest_Unit to mosek 7 -- we'll see if this works

**Comment by whoburg** [comment]
@bqpd, it is my belief that this is running mosek 7, yet we are getting failures on `gpkit.tests.t_model.TestSP_mosek.test_unbounded_debugging`. These are occurring on master on gpkit_Push too.

Any ideas?

**Comment by whoburg** [comment]
minor side issue: @galbramc, https://acdl.mit.edu/csi/job/gpkit_PullRequest_PyLint/1007/buildnode=reynolds timed out?

**Comment by galbramc** [comment]
Your build script finds Mosek 8 as long as it's installed.

```
# Found mosek version 8, installed to /home/jenkins/mosek
```

I have to delete the folder mosek/8. Give me a moment.

**Comment by galbramc** [comment]
test this please

**Comment by galbramc** [comment]
The fact that the build script automatically finds Mosek 8 means that I cannot separate Mosek 8 from gpkit_PullRequest and gpkit_Push. As soon as that folder exist any testing will fail.

**Comment by whoburg** [comment]
understood -- just fired off a gpkit_Push build too

**Comment by whoburg** [comment]
@galbramc thanks for your help. merging this.

File: gpkit/tests/t_constraints.py (hunk 1 above)
Comment (bqpd) [inline gpkit/tests/t_constraints.py]: The previous cost wasn't necessarily not tight; y could have the value of 2 without any issue.

Comment (whoburg) [inline gpkit/tests/t_constraints.py]: got it - thanks
